### ISSUE TITLE
Increase test coverage of test_iutils.py on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8582]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8582|
|Type|enhancement|
|Created|2016-07-10 05:19:57Z|
|Branch|https://github.com/twisted/twisted/tree/8582-rodrigc-iutils-py3|

Most of test_iutils.py is skipped on Python3, even though it was added to twisted/python/dist3.py
The test coverage for this should be increased on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__8582 8582
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8582_rodrigc_iutils_py3 8582-rodrigc-iutils-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468127997358448 1468127997358448
changetime__1468236888366830 1468236888366830
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/312

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa set owner to @rodrigc|
|-|-|

This generally looks good, though I have a few questions I asked inline in the diff that I'd like answered quickly. Mind answering those and then assigning this back to me?

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @Lukasa|
|-|-|

I implemented your feedback

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_4"></a>|@Lukasa set owner to @rodrigc|
|-|-|

LGTM, merge when ready.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub set status to `closed`|
|-|-|

In changeset 09d06350a3b5a18d03ec6a15bf6b8857b70b84a7
```
#!CommitTicketReference repository="" revision="09d06350a3b5a18d03ec6a15bf6b8857b70b84a7"
Merge pull request #312 from twisted/8582-rodrigc-iutils-py3

Author: rodrigc
Reviewer: Lukasa
Fixes: #8582

Increase test coverage of test_utils.py on Python 3
```
